models/safe/zksync/safe_zksync_transactions.sql

```diff
@@ -0,0 +1,112 @@
+{{ 
+    config(
+        materialized='incremental',
+        schema = 'safe_zksync',
+        alias = 'transactions',
+        partition_by = ['block_month'],
+        unique_key = ['block_date', 'tx_hash', 'trace_address'], 
```

Comment:
it looks like this spell is failing on duplicates after the incremental run, which typically means the unique keys are incorrect and/or contain nulls, so the generated merge statement fails to find a match when it should.

can you validate these are the correct unique keys? are we missing a field? if not, then check if any of these contain null values. if there are nulls, you can look to introduce a surrogate key instead of using fields individually, which will never output a null.

here is example of surrogate key, if you get to the point of needing it:
assign in config [like this](https://github.com/duneanalytics/spellbook/blob/main/models/_sector/tokens/arbitrum/tokens_arbitrum_base_transfers.sql#L9)
build [like this](https://github.com/duneanalytics/spellbook/blob/main/macros/models/_sector/tokens/transfers_base.sql#L54)

Reply:
Super helpful, thanks! It's all running fine and the tests are passing now :)